# HG changeset patch
# User xunxun1982
# Date 1411411987 -28800
#      Tue Sep 23 02:53:07 2014 +0800
# Branch MOBILE400b10_2015080313_RELBRANCH
# Node ID 1b0736a6e19113cd117040fdc816418e3ce46b06
# Parent  f015b382bd8dbf95bd66bdaf4943255ad9d72a6f
avx2-fix

diff --git a/media/libvpx/moz.build b/media/libvpx/moz.build
--- a/media/libvpx/moz.build
+++ b/media/libvpx/moz.build
@@ -79,11 +79,21 @@ if CONFIG['CLANG_CL'] or not CONFIG['_MS
                 SOURCES[f].flags += CONFIG['SSE2_FLAGS']
             if 'ssse3' in f:
                 SOURCES[f].flags += ['-mssse3']
             if 'sse4' in f:
                 SOURCES[f].flags += ['-msse4.1']
             if 'avx2' in f:
                 SOURCES[f].flags += ['-mavx2']
 
+if CONFIG['_MSC_VER']:
+    for f in SOURCES:
+        if f.endswith('.c'):
+            if 'sse2' in f:
+                SOURCES[f].flags += ['-arch:SSE2']
+            if 'ssse3' in f:
+                SOURCES[f].flags += ['-arch:SSE2']
+            if 'avx2' in f:
+                SOURCES[f].flags += ['-arch:AVX2']
+
 # Suppress warnings in third-party code.
 if CONFIG['GNU_CC']:
     CFLAGS += ['-Wno-sign-compare']
